soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -187,9 +188,39 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	uint32_t vecbase;
 	uint32_t idc_reg;
 
-	__ASSERT(cpu_num == 1, "Only supports only two CPUs!");
+#ifdef CONFIG_SOC_SERIES_INTEL_CAVS_V25
+	/* On cAVS v2.5, MP startup works differently.  The core has
+	 * no ROM, and starts running immediately upon receipt of an
+	 * IDC interrupt at the start of LPSRAM at 0xbe800000.  Note
+	 * that means we don't need to bother constructing a "message"
+	 * below, it will be ignored.  But it's left in place for
+	 * simplicity and compatibility.
+	 *
+	 * All we need to do is place a single jump at that address to
+	 * our existing MP entry point.  Unfortunately Xtensa makes
+	 * this difficult, as the region is beyond the range of a
+	 * relative jump instruction, so we need an immediate, which
+	 * can only be backwards-referenced.  So we hand-assemble a
+	 * tiny trampoline here ("jump over the immediate address,
+	 * load it, jump to it").
+	 *
+	 * Long term we want to have this in linkable LP-SRAM memory
+	 * such that the standard system bootstrap out of IMR can
+	 * place it there.  But this is fine for now.
+	 */
+	void **lpsram = z_soc_uncached_ptr((void *)LP_SRAM_BASE);
+	uint8_t tramp[] = {
+		0x06, 0x01, 0x00, /* J <PC+8>  (jump to L32R) */
```

Comment:
It's actually the assembler itself; it won't generate a L32R instruction unless you use an assembler generated immediate, which we can't do here because we don't link in lpsram at all.  There's a comment to that effect: what we really want is to have lp-sram be part of the same set of address spaces and visible for linker symbols, and then be copied out of the IMR bootstrap image in the same way.

FWIW: that sequence is really just a jump.  It's the smallest possible "jump to absolute address from anywhere" possible in Xtensa, hilariously.